src/TodosContext.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState } from 'react';
+import { Todo } from './services/types';
+import { Filter } from './services/enums';
+
+interface FilterParams {
+  filterBy?: Filter,
+}
+
+function filterTodosByCompleted(todos: Todo[], filter: Filter): Todo[] {
+  let todosCopy = [...todos];
+
+  switch (filter) {
+    case (Filter.ACTIVE): {
+      todosCopy = todosCopy.filter(todo => !todo.completed);
+      break;
+    }
+
+    case (Filter.COMPLETED): {
+      todosCopy = todosCopy.filter(todo => todo.completed);
+      break;
+    }
+
+    default: {
+      break;
+    }
+  }
+
+  return todosCopy;
+}
+
+function filterTodos(todos: Todo[], { filterBy }: FilterParams): Todo[] {
```

Comment:
would be better to use arrow functions instead of function declaration. Follow some one consistency in your project.
And why do we need to keep filterTodos, filterTodosByCompleted, useLocalStorage inside TodosContext file? move utils to `utils` folder, hooks to `hooks` folder